pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -362,4 +368,216 @@ var _ = Describe("AWSNodeTemplateController", func() {
 			Expect(nodeTemplate.Status.SecurityGroups).To(BeNil())
 		})
 	})
+	Context("AMI Status", func() {
+		BeforeEach(func() {
+			awsEnv.EC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{
+				Images: []*ec2.Image{
+					{
+						Name:         aws.String("test-ami-1"),
+						ImageId:      aws.String("ami-test1"),
+						CreationDate: aws.String(time.Now().Format(time.RFC3339)),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String("test-ami-1")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+					{
+						Name:         aws.String("test-ami-2"),
+						ImageId:      aws.String("ami-test2"),
+						CreationDate: aws.String(time.Now().Add(time.Minute).Format(time.RFC3339)),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String("test-ami-2")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+					{
+						Name:         aws.String("test-ami-3"),
+						ImageId:      aws.String("ami-test3"),
+						CreationDate: aws.String(time.Now().Add(2 * time.Minute).Format(time.RFC3339)),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String("test-ami-3")},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+				},
+			})
+		})
+		It("should resolve amiSelector AMIs and requirements into status", func() {
+			version := lo.Must(awsEnv.AMIProvider.KubeServerVersion(ctx))
+
+			awsEnv.SSMAPI.PresetParameterOutput = map[string]string{
```

Comment:
```suggestion
			awsEnv.SSMAPI.Parameters = map[string]string{
```

Just an idea to shorten the name